vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxView.java

```diff
@@ -57,6 +64,80 @@ protected void initView() {
         configurationForRequired(); // Validation
         themeVariantsHorizontal();// Theme Variants
         styling(); // Styling
+        dataViewDemo();
+    }
+
+    private void dataViewDemo() {
+        final CheckboxGroup<String> checkboxGroup = new CheckboxGroup<>();
+
+        final List<String> items = new ArrayList<>();
+        items.add("Item1");
+        items.add("Item4");
+        items.add("Item2");
+        items.add("Item3");
+
+        ListDataProvider<String> dataProvider = DataProvider.ofCollection(items);
+
+        final CheckboxGroupListDataView<String> dataView =
+                checkboxGroup.setDataProvider(dataProvider);
+
+        Button filterButton = new Button("Filter",
+                event -> dataView.withFilter(item -> item.equals("Item2")));
+
+        Button orderButton = new Button("Order",
+                event -> dataView.withSortComparator(String::compareTo));
+
+        Span size = new Span(String.valueOf(dataView.getDataSize()));
+
+        Button addButton = new Button("Add item", event -> {
+            dataProvider.getItems().add("Item" + new Random().nextInt(10));
+            dataProvider.refreshAll();
+            dataView.getDataSize();
+        });
+
+        Button deleteButton = new Button("Delete item", event -> {
+            dataProvider.getItems().removeAll(checkboxGroup.getValue());
+            dataProvider.refreshAll();
+            dataView.getDataSize();
+        });
+
+        Span containsItem = new Span(String.valueOf(dataView.dataContainsItem("Item1")));
+
+        dataView.addSizeChangeListener(event -> {
+            size.setText(String.valueOf(event.getSize()));
+            containsItem.setText(String.valueOf(dataView.dataContainsItem("Item1")));
+            System.out.println(dataView.getAllItems().collect(Collectors.toList()));
+        });
+
+        Span currentItemSpan = new Span("Item1");
+
+        Span hasItemSpan = new Span("--");
+
+        final String[] currentItem = {"Item1"};
```

Comment:
Instead of a String[] use AtomicReference<String> for a cleaner read.

e.g.
```java
AtomicReference<String> currentItem = new AtomicReference<>("item1");
...
event -> {
  dataView.getNextItem(currentItem.get());
  currentItem.set(nextItem);
  ...
}
```

Reply:
Thanks! That's much better. I used a tip from IDEA to be honest.

Reply:
Done.